sdk/storage/Azure.Storage.Queues/tests/QueueClientTests.cs

```diff
@@ -1977,5 +1977,55 @@ public void CanMockClientConstructors()
             mock = new Mock<QueueClient>(new Uri("https://test/test"), new AzureSasCredential("foo"), new QueueClientOptions()).Object;
             mock = new Mock<QueueClient>(new Uri("https://test/test"), mockTokenCredential, new QueueClientOptions()).Object;
         }
+
+        [RecordedTest]
+        public void QueueClientPolicyPermissions_checkIfSetCorrectly()
+        {
+            // Test #1
+            string permissionsStr1 = "ra";
+            QueueAccessPolicyPermissions permissionsEnum1 = QueueAccessPolicyPermissions.Read | QueueAccessPolicyPermissions.Add;
+
+            QueueAccessPolicy qap = new QueueAccessPolicy
+            {
+                Permissions = permissionsStr1
+            };
+
+            Assert.AreEqual(qap.Permissions, permissionsStr1);
+            Assert.AreEqual(qap.QueueAccessPolicyPermissions.ToPermissionsString(), permissionsEnum1.ToPermissionsString());
+
+            // Test #2
+            string input2 = "paru";
+            string permissionsStr2 = "raup";
```

Comment:
Instead of having 4 tests in 1 test. It will probably be better to have a use `[TestCase(<permissions>)` instead.

See
https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/storage/Azure.Storage.Blobs/tests/AppendBlobClientTests.cs#L1343-L1347

for an example how to do this.